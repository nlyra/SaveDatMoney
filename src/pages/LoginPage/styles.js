import { Platform, StyleSheet } from 'react-native';

export default StyleSheet.create({
    container: {
        flex: 1,
        alignSelf: "center",
        width: '100%',
        ...Platform.select({
            ios: {
                width: '100%'
            },
            android: {
                width: '100%'
            },
            default: {
              // other platforms, web for example
              width: '25%'
            }
          }),
    },
    title: {
        alignItems: 'center'
    },
    logo: {
        flex: 1,
        height: 160,
        width: 140,
        alignSelf: "center",
        margin: 30
    },
    input: {
        height: 48,
        borderRadius: 5,
        overflow: 'hidden',
        backgroundColor: 'white',
        marginTop: 10,
        marginBottom: 10,
        marginLeft: 30,
        marginRight: 30,
        paddingLeft: 16
    },
    button: {
        backgroundColor: '#2ea44f',
        marginLeft: 30,
        marginRight: 30,
        marginTop: 20,
        height: 48,
        borderRadius: 5,
        alignItems: "center",
        justifyContent: 'center'
    },
    buttonTitle: {
        color: 'white',
        fontSize: 16,
        fontWeight: "bold"
    },
    footerView: {
        flex: 1,
        alignItems: "center",
        marginTop: 10
    },
    footerText: {
        fontSize: 16,
        color: '#2e2e2d'
    },
    footerLink: {
        color: "#2ea44f",
        fontWeight: "bold",
        fontSize: 16
    }
})
