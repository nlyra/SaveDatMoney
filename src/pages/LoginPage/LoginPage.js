import React, { useState } from 'react'
import { Image, Text, TextInput, TouchableOpacity, View } from 'react-native'
import { ScrollView } from 'react-native-gesture-handler';
import styles from './styles';
import { colors, padding, fonts, buttons } from '../stdStyles';

import { firebase } from '../../firebase/config'

export default function LoginPage({navigation})
{
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [message, setMessage] = useState('');

    const onFooterLinkPress = () => {
        navigation.navigate('Registration')
    }
    const onFooterLinkPress2 = () => {
        navigation.navigate('ForgotPassword')
    }

    /**
     *
     * @param {React.FormEvent<HTMLFormElement>} e
     */
    const doLogin = (e) => {

        e.preventDefault();

        firebase.auth().signInWithEmailAndPassword(email, password).then((response) => {

            const uid = response.user.uid
            const usersRef = firebase.database().ref('users/' + uid);

            usersRef.once('value').then(function(snapshot) {
                navigation.reset({
                  index: 0,
                  routes: [{ name: 'Budget' }],
                });
                //navigation.navigate('Dashboard', snapshot.val());
            }).catch((error) => {
                var errorCode = error.code;
                var errorMessage = error.message;
                setMessage(errorMessage);
            })
        }).catch((error) => {
            var errorCode = error.code;
            var errorMessage = error.message;
            setMessage(errorMessage);
        })
    }

    return (
        <View style={styles.container}>
            <ScrollView style={{ flex: 1, width: '100%' }} keyboardShouldPersistTaps="always">
                <Image
                    style={styles.logo}
                    source={require('../../../assets/savedatmoneylogo.png')}
                />
                <TextInput
                    style={styles.input}
                    placeholder='E-mail'
                    placeholderTextColor="#aaaaaa"
                    onChangeText={(text) => setEmail(text)}
                    value={email}
                    underlineColorAndroid="transparent"
                    autoCapitalize="none"
                />
                <TextInput
                    style={styles.input}
                    placeholderTextColor="#aaaaaa"
                    secureTextEntry
                    placeholder='Password'
                    onChangeText={(text) => setPassword(text)}
                    value={password}
                    underlineColorAndroid="transparent"
                    autoCapitalize="none"
                />
                {message ? <Text style={{ color: 'red' }}>{message}</Text> : null}
                <TouchableOpacity style={styles.button} onPress={doLogin}>
                    <Text style={styles.buttonTitle}>Log in</Text>
                </TouchableOpacity>
                <View style={styles.footerView}>
                    <Text style={styles.footerText}>Don't have an account? <Text onPress={onFooterLinkPress} style={styles.footerLink}>Sign up</Text></Text>
                </View>
                <View style={styles.footerView}>
                    <Text onPress={onFooterLinkPress2} style={styles.footerLink}>Forgot Password?</Text>
                </View>
            </ScrollView>
        </View>
    )
}
