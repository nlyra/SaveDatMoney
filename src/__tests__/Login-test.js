import React from 'react';
import renderer from 'react-test-renderer';
import LoginPage from '../pages/LoginPage/LoginPage';

test('renders correctly', () => {
    const tree = renderer.create(<LoginPage />).toJSON();
    expect(tree).toMatchSnapshot();
})