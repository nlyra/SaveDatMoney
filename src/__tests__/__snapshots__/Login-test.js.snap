// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    keyboardShouldPersistTaps="always"
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../assets/savedatmoneylogo.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "height": 160,
            "margin": 30,
            "width": 140,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="E-mail"
        placeholderTextColor="#aaaaaa"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 5,
            "height": 48,
            "marginBottom": 10,
            "marginLeft": 30,
            "marginRight": 30,
            "marginTop": 10,
            "overflow": "hidden",
            "paddingLeft": 16,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#aaaaaa"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 5,
            "height": 48,
            "marginBottom": 10,
            "marginLeft": 30,
            "marginRight": 30,
            "marginTop": 10,
            "overflow": "hidden",
            "paddingLeft": 16,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2ea44f",
            "borderRadius": 5,
            "height": 48,
            "justifyContent": "center",
            "marginLeft": 30,
            "marginRight": 30,
            "marginTop": 20,
            "opacity": 1,
            "paddingLeft": 30,
            "paddingRight": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Log in
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2e2e2d",
              "fontSize": 16,
            }
          }
        >
          Don't have an account? 
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#2ea44f",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Sign up
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginTop": 10,
          }
        }
      >
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#2ea44f",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Forgot Password?
        </Text>
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderRadius": 20,
              "elevation": 9,
              "margin": 20,
              "padding": 35,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 16,
                "paddingLeft": 20,
              }
            }
          >
            Please verifiy your email before signing in.
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 20,
                "paddingLeft": 30,
                "paddingRight": 30,
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2ea44f",
                  "borderRadius": 5,
                  "height": 48,
                  "justifyContent": "center",
                  "marginLeft": 30,
                  "marginRight": 30,
                  "marginTop": 20,
                  "paddingLeft": 30,
                  "paddingRight": 30,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Resend Link
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2ea44f",
                  "borderRadius": 5,
                  "height": 48,
                  "justifyContent": "center",
                  "marginLeft": 30,
                  "marginRight": 30,
                  "marginTop": 20,
                  "paddingLeft": 30,
                  "paddingRight": 30,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Cancel
              </Text>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </RCTScrollView>
</View>
`;
